### PR TITLE
feat(autoenv): add path for Apple Silicon Homebrew

### PR DESCRIPTION
**Pull request by jmeridth**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.

## Changes:

- [x] look for the plugin in /opt/homebrew/opt - Apple silicon default path for homebrew

## Other comments:

Based the path on my local testing and from Homebrew's own installation page [here](https://docs.brew.sh/Installation)

```
This script installs Homebrew to its preferred prefix 
(/usr/local for macOS Intel, 
/opt/homebrew for Apple Silicon and 
/home/linuxbrew/.linuxbrew for Linux)
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,6 @@ cache/
 log/
 *.swp
 .DS_Store
+
```
Comment (carlosala) [inline .gitignore:9]: Please, remove the changes in `.gitignore`, there are no ide-specific folders there!

File: plugins/autoenv/autoenv.plugin.zsh [hunk 2]
```diff
@@ -13,6 +13,7 @@ if ! type autoenv_init >/dev/null; then
       ~/.autoenv
       ~/.local/bin
       /usr/local/opt/autoenv
+      /opt/homebrew/opt/autoenv
```
Comment (carlosala) [inline plugins/autoenv/autoenv.plugin.zsh:16]: Looks nice, we have done some changes to support M1 brew folder structure, for example in #10618, so this looks as a natural continuation of that.

File: .gitignore (hunk 1 above)
Comment (jmeridth) [reply .gitignore]: Done

**Comment by jmeridth** [comment]
@carlosala changes made.  thank you for reviewing.

**Review by carlosala — APPROVED** [review]

**Comment by jmeridth** [comment]
Thank you @carlosala and @mcornella.  Cheers 🍻

**Comment by apjanke** [comment]
Sorry for the late comment here! I'm not up on the Apple Silicon side of things to have anything useful to say about this, anyway. Cheers!

**Pull request merged**